www/src/views/user.ts

```diff
@@ -3,62 +3,118 @@ import base from './base';
 import Network from '../services/network';
 import Model from '../models/statistics';
 import Chart from 'chart.js';
+import {UserModel} from '../models/user';
 
 class UserProfile implements ξ.ClassComponent {
-  view(vnode: ξ.CVnode) {
-    const username = vnode.attrs.username;
-    const fullname = vnode.attrs.fullname;
-    const email = vnode.attrs.email;
+    isFriend: boolean = false;
 
-    return ξ(base,
-        ξ('.ui.text.container', {
-          style: 'margin-top: 1em;',
-        },
-        ξ('h1.ui.header',
-            ξ('img.ui.circular.image', {
-              src: `/user/${username}/avatar`,
-            }),
-            ξ('.content', `${username}\'S PROFILE`.toUpperCase(),
-                ξ('.sub.header', `You are viewing ${fullname}'s profile.`)),
-        ),
-        ξ('.ui.divider')),
-        ξ('.ui.main.text.container.two.column.stackable.grid',
-            ξ('.ui.column',
-                ξ('img.ui.image.rounded.medium#picture',
-                    {src: '/user/' + username + '/avatar'},
-                    'User profile picture.'),
-                ξ('h2',
-                    fullname),
-                ξ('p',
-                    ξ('i.user.icon'),
-                    username),
-                ξ('p',
-                    ξ('i.mail.icon'),
-                    email),
-                ξ('.ui.placeholder',
-                    ξ('ui.image')),
+    hasFriend(username: string) {
+      for (const friend of UserModel.friends) {
+        if (friend.friend == username) {
+          this.isFriend = true;
+          break;
+        } else {
+          this.isFriend = false;
+        }
```

Comment:
This could be shortened to:
```ts
this.isFriend = false;
for (const friend of UserModel.friends) {
  if (friend.friend == username) {
    this.isFriend = true;
    break;
  }
}
```
Oooooooooor:
```ts
UserModels.friends.map(({friend}) => { if (friend == username) { this.isFriend = true; } });
```

Don't do the second, I was bored

Reply:
LES OPTIONALES